src/framework/theme/components/search/search.component.ts

```diff
@@ -320,6 +329,10 @@ export class NbSearchComponent implements OnInit, OnDestroy {
     this.hideSearch();
   }
 
+  searchOnInput(term: string) {
```

Comment:
Let's change `searchOnInput` to `emitInput` to be more descriptive about what this method actually does.